src/test/trackmetadataexport_test.cpp

```diff
@@ -0,0 +1,138 @@
+#include <gtest/gtest.h>
+
+#include <QTemporaryDir>
+
+#include "test/mixxxtest.h"
+#include "test/soundsourceproviderregistration.h"
+#include "track/globaltrackcache.h"
+#include "track/track.h"
+
+namespace {
+
+const QString kEmptyFile = QStringLiteral("empty.mp3");
+
+void deleteTrack(Track* pTrack) {
+    // Delete track objects directly in unit tests with
+    // no main event loop
+    delete pTrack;
+};
+
+class GlobalTrackCacheHelper : public GlobalTrackCacheSaver {
+  public:
+    void saveEvictedTrack(Track* pTrack) noexcept override {
+        ASSERT_FALSE(pTrack == nullptr);
+    }
+    GlobalTrackCacheHelper() {
+        GlobalTrackCache::createInstance(this, deleteTrack);
+    }
+    ~GlobalTrackCacheHelper() override {
+        GlobalTrackCache::destroyInstance();
+    }
+};
+
+} // namespace
+
+class TrackMetadataExportTest : public MixxxTest, SoundSourceProviderRegistration {
```

Comment:
you'll need access specifiers for each Base class. currently `SoundSourceProviderRegistration` is `private` (by default). Is that intentional?
```suggestion
class TrackMetadataExportTest : public MixxxTest, private SoundSourceProviderRegistration {
```